rust/cyclotron-node/src/lib.rs

```diff
@@ -422,6 +469,144 @@ fn set_parameters(mut cx: FunctionContext) -> JsResult<JsNull> {
     Ok(cx.null())
 }
 
+fn set_blob(mut cx: FunctionContext) -> JsResult<JsNull> {
+    let arg = cx.argument::<JsString>(0)?.value(&mut cx);
+    let job_id: Uuid = arg
+        .parse()
+        .or_else(|_| cx.throw_error(format!("invalid job id: {}", arg)))?;
+
+    // Tricky - we have to support passing nulls here, because that's how you clear the blob.
+    let blob = cx.argument::<JsValue>(1)?;
+    let blob: Option<Vec<u8>> =
+        if blob.is_a::<JsNull, _>(&mut cx) || blob.is_a::<JsUndefined, _>(&mut cx) {
+            None
+        } else {
+            Some(
+                blob.downcast_or_throw::<JsUint8Array, _>(&mut cx)?
+                    .as_slice(&cx)
+                    .to_vec(),
+            )
+        };
+
+    WORKER
+        .get()
+        .map_or_else(|| cx.throw_error("worker not initialized"), Ok)?
+        .set_blob(job_id, blob)
+        .or_else(|e| cx.throw_error(format!("{}", e)))?;
+
+    Ok(cx.null())
+}
+
+fn jobs_to_js_array<'a>(cx: &mut TaskContext<'a>, jobs: Vec<Job>) -> JsResult<'a, JsArray> {
```

Comment:
This was a pain, but we should probably do it for the incoming (`createJob`) side too, eventually. Paying to JSON ser/de seems like a waste.

Reply:
I'm kind of coming around to "dequeue_job" returning /only/ the job ID, and then if you want to access any other member of it you have to ask the worker (this lets us abstract over vm state/blob storage, which might be _really_ useful in the future). Internally right now the worker would just hold the job in a hashmap, and the basic dequeue operation would still return the whole row. We can talk about it more in sync, and for now I think this is fine.